packages/eslint-plugin-template/src/rules/prefer-ngsrc.ts

```diff
@@ -66,3 +69,13 @@ function hasNormalSrcAttribute({
 }: TmplAstElement): TmplAstTextAttribute | TmplAstBoundAttribute | undefined {
   return [...inputs, ...attributes].find(({ name }) => name === 'src');
 }
+
+function isSrcBase64Image(
+  attribute: TmplAstTextAttribute | TmplAstBoundAttribute,
+) {
+  if (attribute instanceof TmplAstTextAttribute) {
+    return attribute.value.trim().startsWith('data:');
```

Comment:
I'm not an expert on Base64, but would it be worth adding a check for the value to also contain `;base64`?

Reply:
I apologize for the lack of details! This fix follows angular's assertion[^1] that ngSrc does not contain `data:`. So it seems that not only is base64 not allowed but any data URI is not allowed, including svg that are **NOT** base64 encoded.

Example[^2]:
```<img ngSrc="data:image/svg+xml;charset=UTF-8,<svg xmlns='http://www.w3.org/2000/svg'><circle r='50' cx='50' cy='50' fill='red'/></svg>">```

Reference:
[^1]: https://github.com/angular/angular/blob/17.0.3/packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts#L585

[^2]: https://stackblitz.com/edit/angular-qrkion?file=src%2Fmain.ts

Reply:
Thank you for clarifying. LGTM, though there is a CI failure [here](https://github.com/angular-eslint/angular-eslint/actions/runs/6924924361/job/18854030056?pr=1628#step:10:490), if you could please take a look.

Also, if this PR fixes #1543 , could you please update the PR description to automatically close the issue upon merge? See [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).